x-pack/plugins/security_solution/public/management/components/endpoint_policy_link.tsx

```diff
@@ -6,60 +6,200 @@
-  const clickHandler = useNavigateByRouterEventHandler(toRoutePath);
+export const EndpointPolicyLink = memo<EndpointPolicyLinkProps>(
+  ({
+    policyId,
+    backLink,
+    children,
+    policyExists = true,
+    isOutdated = false,
+    tooltip = true,
+    revision,
+    textSize = 's',
+    ...euiLinkProps
+  }) => {
+    const { getAppUrl } = useAppUrl();
+    const { canReadPolicyManagement } = useUserPrivileges().endpointPrivileges;
+    const testId = useTestIdGenerator(euiLinkProps['data-test-subj']);
 
-  if (!policyId || missingPolicies[policyId]) {
-    return (
-      <span className={otherProps.className} data-test-subj={otherProps['data-test-subj']}>
-        {children}
-        {
-          <EuiText color="subdued" size="xs" className="eui-textNoWrap">
+    const { toRoutePath, toRouteUrl } = useMemo(() => {
+      const path = policyId ? getPolicyDetailPath(policyId) : '';
+      return {
+        toRoutePath: backLink ? { pathname: path, state: { backLink } } : path,
+        toRouteUrl: getAppUrl({ path }),
+      };
+    }, [policyId, getAppUrl, backLink]);
+
+    const clickHandler = useNavigateByRouterEventHandler(toRoutePath);
+
+    const displayAsLink = useMemo(() => {
+      return Boolean(canReadPolicyManagement && policyId && policyExists);
+    }, [canReadPolicyManagement, policyExists, policyId]);
+
+    const displayValue = useMemo(() => {
+      const content = (
+        <EuiText
+          className="eui-displayInline eui-textTruncate"
+          size={textSize}
+          data-test-subj={testId('displayContent')}
+        >
+          {children}
+        </EuiText>
+      );
+
+      return displayAsLink ? (
+        // eslint-disable-next-line @elastic/eui/href-or-on-click
+        <EuiLink
+          href={toRouteUrl}
+          onClick={clickHandler}
+          {...euiLinkProps}
+          data-test-subj={testId('link')}
+        >
+          {content}
+        </EuiLink>
+      ) : (
+        content
+      );
+    }, [children, clickHandler, displayAsLink, euiLinkProps, testId, textSize, toRouteUrl]);
+
+    const policyNoLongerAvailableMessage = useMemo(() => {
+      return (
+        ((!policyId || !policyExists) && (
+          <EuiText
+            color="subdued"
+            size="xs"
+            className="eui-textNoWrap"
+            data-test-subj={testId('policyNotFoundMsg')}
+          >
             <EuiIcon size="m" type="warning" color="warning" />
             &nbsp;
-            <FormattedMessage
-              id="xpack.securitySolution.endpoint.policyNotFound"
-              defaultMessage="Policy not found!"
-            />
+            {POLICY_NOT_FOUND_MESSAGE}
           </EuiText>
-        }
-      </span>
-    );
-  }
+        )) ||
+        null
+      );
+    }, [policyExists, policyId, testId]);
+
+    const tooltipContent: React.ReactNode | undefined = useMemo(() => {
+      const content = tooltip === true ? children : tooltip || undefined;
```

Comment:
I wonder if this couldn't be simplified to:
 ```
 const content = tooltip === true ? children : undefined;
 ```
 What is the reason behind `: tooltip` part ?

Reply:
what if `tooltip` is an empty string?  In that case, we don't want the tooltip to just show up with no info. the `|| undefined` handles that condition.